inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/InstrumentationConfigurationResolver.java

```diff
@@ -91,6 +90,31 @@ public ClassInstrumentationConfiguration getClassInstrumentationConfiguration(Cl
         }
     }
 
+    public Map<MethodDescription, MethodHookConfiguration> getHookConfigurations(Class<?> clazz) {
+        val config = currentConfig;
+        TypeDescription type = TypeDescription.ForLoadedType.of(clazz);
+        Set<InstrumentationRule> narrowedRules = getNarrowedRulesFor(type, config);
+
+        Set<InstrumentationScope> matchesScopes = narrowedRules.stream()
+                .flatMap(r -> r.getScopes().stream())
+                .distinct()
+                .filter(scope -> scope.getTypeMatcher().matches(type))
+                .collect(Collectors.toSet());
+
+        Map<MethodDescription, MethodHookConfiguration> result = new HashMap<>();
+        if (!matchesScopes.isEmpty()) {
+            for (val method : type.getDeclaredMethods()) {
+                val matchedScopes = matchesScopes.stream().filter(scope -> scope.getMethodMatcher().matches(method)).collect(Collectors.toSet());
+                val matchedRules = narrowedRules.stream().filter(r -> !Collections.disjoint(r.getScopes(), matchedScopes)).collect(Collectors.toSet());
```

Comment:
I am not understanding this line? can you add a small explanation here.. maybe `matchesScopes` should be renamed to `matchedTypeScopes` and `matchedScopes` to `matchedMethodScopes`, this way it;s so confusing?